project-base/app/tests/FrontendApiBundle/Functional/Product/Flag/FlagTest.php

```diff
@@ -109,90 +111,82 @@ public function testFlagByUuidFilteredByAnotherFlag(): void
-                        {
-                            "name": "' . t('Printers', [], Translator::DATA_FIXTURES_TRANSLATION_DOMAIN, $this->getFirstDomainLocale()) . '"
-                        },
-                        {
-                            "name": "' . t('Books', [], Translator::DATA_FIXTURES_TRANSLATION_DOMAIN, $this->getFirstDomainLocale()) . '"
-                        },
-                        {
-                            "name": "' . t('Mobile Phones', [], Translator::DATA_FIXTURES_TRANSLATION_DOMAIN, $this->getFirstDomainLocale()) . '"
-                        },
-                        {
-                            "name": "' . t('Food', [], Translator::DATA_FIXTURES_TRANSLATION_DOMAIN, $this->getFirstDomainLocale()) . '"
-                        }
-                    ]
-                }
-            }
-        }';
+        $products = [
+            [
+                'name' => t('22" Sencor SLE 22F46DM4 HELLO KITTY', [], Translator::DATA_FIXTURES_TRANSLATION_DOMAIN, $this->getFirstDomainLocale()),
+            ],
+            [
+                'name' => t('32" Philips 32PFL4308', [], Translator::DATA_FIXTURES_TRANSLATION_DOMAIN, $this->getFirstDomainLocale()),
+            ],
+            [
+                'name' => t('32" Hyundai 32PFL4400', [], Translator::DATA_FIXTURES_TRANSLATION_DOMAIN, $this->getFirstDomainLocale()),
+            ],
+            [
+                'name' => t('Apple iPhone 5S 64GB, gold', [], Translator::DATA_FIXTURES_TRANSLATION_DOMAIN, $this->getFirstDomainLocale()),
+            ],
+            [
+                'name' => t('Book 55 best programs for burning CDs and DVDs', [], Translator::DATA_FIXTURES_TRANSLATION_DOMAIN, $this->getFirstDomainLocale()),
+            ],
+            [
+                'name' => t('Book Computer for Dummies Digital Photography II', [], Translator::DATA_FIXTURES_TRANSLATION_DOMAIN, $this->getFirstDomainLocale()),
+            ],
+            [
+                'name' => t('Book of procedures for dealing with traffic accidents', [], Translator::DATA_FIXTURES_TRANSLATION_DOMAIN, $this->getFirstDomainLocale()),
+            ],
+            [
+                'name' => t('Canon MG3550', [], Translator::DATA_FIXTURES_TRANSLATION_DOMAIN, $this->getFirstDomainLocale()),
+            ],
+            [
+                'name' => t('Defender 2.0 SPK-480', [], Translator::DATA_FIXTURES_TRANSLATION_DOMAIN, $this->getFirstDomainLocale()),
+            ],
+            [
+                'name' => t('DeLonghi ECAM 44.660 B Eletta Plus', [], Translator::DATA_FIXTURES_TRANSLATION_DOMAIN, $this->getFirstDomainLocale()),
+            ],
+        ];
 
-        $this->assertQueryWithExpectedJson($query, $jsonExpected);
+        ArraySorter::sortArrayAlphabeticallyByValue('name', $products);
+
+        $productsWithNodes = [];
+
+        for ($i = 0; $i < $limit; $i++) {
+            $productsWithNodes[] = [
+                'node' => $products[$i],
+            ];
+        }
```

Comment:
why is necessary to prepare 10 products and then use only 5?

Reply:
Again - I hope that the first ten will be sufficient for the test to work in most of the languages.